diff --git AndroidBillingLibrary/src/net/robotmedia/billing/BillingRequest.java AndroidBillingLibrary/src/net/robotmedia/billing/BillingRequest.java
index 8362559..f18f55f 100644
--- AndroidBillingLibrary/src/net/robotmedia/billing/BillingRequest.java
+++ AndroidBillingLibrary/src/net/robotmedia/billing/BillingRequest.java
@@ -217,10 +217,8 @@ public abstract class BillingRequest {
     	@Override
     	public void onResponseCode(ResponseCode response) {
     		super.onResponseCode(response);
-    		if (response == ResponseCode.RESULT_OK) {
     			BillingController.onTransactionsRestored();
     		}
-    	}
     	
     }
 
